### PR TITLE
fix : [CDS-46164]: Adding logs for service step in console for custom artifact step

### PR DESCRIPTION
**Pull request by Y0709**
Added console logs on the delegate side. Also added check for harness variable at the end for the custom artifact script

Screenshot 2023-03-01 at 2 34 24 AM

- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/43563)

### CODE REVIEW

File: 120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/util/ArtifactResourceUtils.java [hunk 1]
```diff
@@ -774,12 +774,15 @@ && isNotEmpty(customScriptInlineSource.getScript().fetchFinalValue().toString())
             customArtifactConfig.getScripts().getFetchAllArtifacts().getVersionPath().fetchFinalValue().toString();
       }
     }
+    if (!script.endsWith("$HARNESS_ARTIFACT_RESULT_PATH")) {
+      throw new io.harness.exception.HintException("Artifact path can only be HARNESS_ARTIFACT_RESULT_PATH");
+    }
```
Comment (shivamnegi94) [inline 120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/util/ArtifactResourceUtils.java:779]: Improve the error message here

Comment (shivamnegi94) [reply 120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/util/ArtifactResourceUtils.java:779]: please add the UT for this

Comment (Y0709) [reply 120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/util/ArtifactResourceUtils.java:785]: done

**Review by shivamnegi94 — DISMISSED** [review]

**Review by shivamnegi94 — COMMENTED** [review]
> Reviewed 5 of 9 files at r1, 8 of 9 files at r2, 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43563)* status: 15 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Comment by Y0709** [comment]
trigger ti3

**Comment by yogesh-chauhan** [comment]
*[`120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/util/ArtifactResourceUtils.java` line 784 at r4](https://reviewable.io/reviews/harness/harness-core/43563#-NNfvEQDBTplpRGCTmL4:-NNfvEQDBTplpRGCTmL5:b-9o56ug) ([raw file](https://github.com/harness/harness-core/blob/8bc5a8babc837ce23b28c87ba03ea7a2dd44975a/120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/util/ArtifactResourceUtils.java#L784)):*
> ```Java
>     if (!script.endsWith("$HARNESS_ARTIFACT_RESULT_PATH")) {
>       throw new io.harness.exception.HintException(
>           "Invalid Harness variable at the end of the script. Harness variable should be $HARNESS_ARTIFACT_RESULT_PATH");
> ```

why is it mandatory that the script should end with this ? 
```
echo "{..some json} > $HARNESS_ARTIFACT_RESULT_PATH"
echo "hello"

```
isn't this a valid script ?

**Comment by yogesh-chauhan** [comment]
*[`120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/util/ArtifactResourceUtils.java` line 788 at r4](https://reviewable.io/reviews/harness/harness-core/43563#-NNfvRa72NICrjx5VG9O:-NNfvRa72NICrjx5VG9P:b-508hvz) ([raw file](https://github.com/harness/harness-core/blob/8bc5a8babc837ce23b28c87ba03ea7a2dd44975a/120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/util/ArtifactResourceUtils.java#L788)):*
> ```Java
> 
>     if (isEmpty(arrayPath) || NGExpressionUtils.isRuntimeField(arrayPath)) {
>       throw new io.harness.exception.HintException("Array path can not be empty");
> ```

lets not do fully qualified import for this

**Comment by yogesh-chauhan** [comment]
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/artifact/steps/ArtifactsStepV2.java` line 418 at r4](https://reviewable.io/reviews/harness/harness-core/43563#-NNfvYG9EPy4HK-oWJht:-NNfvYGA10RU4u_nNBwT:b-t9nrh4) ([raw file](https://github.com/harness/harness-core/blob/8bc5a8babc837ce23b28c87ba03ea7a2dd44975a/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/artifact/steps/ArtifactsStepV2.java#L418)):*
> ```Java
>         + taskParameters.getArtifactTaskType().getDisplayName();
> 
>     boolean withLogs = artifactConfig.getSourceType().equals(ArtifactSourceType.CUSTOM_ARTIFACT);
> ```

constant should be on the left side of the comparison

**Comment by yogesh-chauhan** [comment]
@yuvrajharness can you post a screenshot of how the logs are looking, on the UI ?

**Review by yogesh-chauhan — COMMENTED** [review]
> Reviewed 4 of 9 files at r1, 6 of 9 files at r2, 1 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43563)* status: 4 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Pull request closed without merging**